### ISSUE TITLE
No events received when using mysql-binlog-connector-java and exception thrown

### ISSUE THREAD

**Issue by nizamk** [shyiko/mysql-binlog-connector-java#92]
Hi,

I am using this package but when i ran the following code i got exception thrown:

	public static void main (String[] args){

		BinaryLogClient client = new BinaryLogClient("127.0.0.1", 3306, "root","new-password");
		client.registerLifecycleListener(new BinaryLogClient.LifecycleListener() {
			@Override
			public void onConnect(BinaryLogClient binaryLogClient) {
				System.out.println("OnConnect()");
			}
			@Override
			public void onCommunicationFailure(BinaryLogClient binaryLogClient, Exception e) {
				System.out.println("OnCommunicationFailure()");
				e.printStackTrace();
			}
			@Override
			public void onEventDeserializationFailure(BinaryLogClient binaryLogClient, Exception e) {
				System.out.println("OnEventDeserialize()");
			}
			@Override
			public void onDisconnect(BinaryLogClient binaryLogClient) {
				System.out.println("OnDisconnect()");
			}
		});
		client.registerEventListener(new BinaryLogClient.EventListener() {
			@Override
			public void onEvent(Event event) {
				System.out.println("event received: " + event);
			}
		});
		try {
			client.connect();
		} catch (IOException e) {
			e.printStackTrace();
		}
		System.out.println("exiting..");

the exceptions:

Apr 01, 2016 1:35:14 AM com.github.shyiko.mysql.binlog.BinaryLogClient connect
INFO: Connected to 127.0.0.1:3306 at mariadb-bin.000940/369 (sid:65535, cid:199)
OnConnect()
com.github.shyiko.mysql.binlog.network.ServerException: Misconfigured master - server id was not set
	at com.github.shyiko.mysql.binlog.BinaryLogClient.listenForEventPackets(BinaryLogClient.java:621)
	at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:405)
	at replicate.Main.main(Main.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
OnCommunicationFailure()
OnDisconnect()
exiting..

Further I was wondering why the call: client.connect() did not block. The program just exited.

I ran command:  mysql -u root -pnew-password -h 127.0.0.1 --port=3306 --protocol=TCP -e 'show master status' and got the following:

+--------------------+----------+--------------+------------------+
| File               | Position | Binlog_Do_DB | Binlog_Ignore_DB |
+--------------------+----------+--------------+------------------+
| mariadb-bin.000940 |      369 |              |                  |
+--------------------+----------+--------------+------------------+

Note: I have read the issue in https://github.com/shyiko/mysql-binlog-connector-java/issues/10 but i still could not figure out my issue.

Really appreciate your help. Not sure how else to debug.

Thanks.

**Comment by shyiko** [comment]
Hi. see #7 (http://dev.mysql.com/doc/refman/5.0/en/replication-options-master.html). 

**On the master and each slave**, you must use the server-id option to establish a unique replication ID. For each server, you should pick a unique positive integer in the range from 1 to 232 − 1, and each ID must be different from every other ID in use by any other replication master or slave. Example: server-id=3.

tldr; check that server-id is set in my.conf.

> Further I was wondering why the call: client.connect() did not block 

`connect` has failed before BINLOG_DUMP was issued. It did block (it just everything went quick enough to give an impression that it didn't).

**Comment by nizamk** [comment]
Hi,

I got it to work! I set the server-id=1 in /etc/mysql/my.cnf.

Then I received the events as expected. It works great! 

event received: Event{header=EventHeaderV4{timestamp=1459475978000, eventType=QUERY, serverId=1, headerLength=19, dataLength=122, nextPosition=535, flags=0}, data=QueryEventData{threadId=33, executionTime=0, errorCode=0, database='demo', sql='insert into users (name,email,department) values ("fon","fon@test.com","engr")'}}
event received: Event{header=EventHeaderV4{timestamp=1459475978000, eventType=XID, serverId=1, headerLength=19, dataLength=8, nextPosition=562, flags=0}, data=XidEventData{xid=102}}
event received: Event{header=EventHeaderV4{timestamp=1459476155000, eventType=QUERY, serverId=1, headerLength=19, dataLength=19, nextPosition=600, flags=8}, data=QueryEventData{threadId=0, executionTime=0, errorCode=0, database='', sql='BEGIN'}}
event received: Event{header=EventHeaderV4{timestamp=1459476155000, eventType=QUERY, serverId=1, headerLength=19, dataLength=101, nextPosition=720, flags=0}, data=QueryEventData{threadId=33, executionTime=0, errorCode=0, database='demo', sql='update users set email="mokiki@abc.com" where name="kiki"'}}

Great work. Well done.

Thanks
